### PR TITLE
Remove old numpy 1.16 work-arounds

### PR DESCRIPTION
**Pull request by mhvk**
Factored out from #11127 to try to make that more manageable (where it is useful as `erfa.ufunc.s2p` becomes much easier to handle for masked arrays/quantities).

But mostly the work-around simply should be removed, since our minimum numpy is 1.17. Sot, just a question of making sure the tests pass!

### CODE REVIEW

**Comment by mhvk** [comment]
The codecov/project is a red herring (in contrast to codecov/patch which brought out uncovered code including a bug...)

**Comment by mhvk** [comment]
@pllim - would you have a chance to review? This PR is really just about removing some numpy 1.16 work-arounds, so I think it can go in...

**Review by pllim — APPROVED** [review]
Tests pass, so LGTM. Thanks!

I'll leave it for you to merge in case you are still waiting for another review.

**Comment by mhvk** [comment]
Thanks, I'll merge as I'm hoping to rebase #11127 on that to make it affect less files.

**Pull request merged**
